include/crypto/ml_kem.h

```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Copyright (c) 2024, Google Inc. */
+
+#ifndef OPENSSL_HEADER_ML_KEM_H
+# define OPENSSL_HEADER_ML_KEM_H
+
+# ifndef OPENSSL_NO_ML_KEM
+
+#  include <openssl/e_os2.h>
+#  include <crypto/evp.h>
+#  include <crypto/ml_kem_types.h>
+
+typedef struct ossl_ml_kem_ctx_st ossl_ml_kem_ctx;
+
+/* General ctx functions */
+ossl_ml_kem_ctx *ossl_ml_kem_newctx(OSSL_LIB_CTX *libctx,
+                                    const char *properties);
+ossl_ml_kem_ctx *ossl_ml_kem_ctx_dup(ossl_ml_kem_ctx *ctx);
+void ossl_ml_kem_ctx_free(ossl_ml_kem_ctx *ctx);
+
+#  define ML_KEM_DEGREE 256
+/*
+ * With (q-1) an odd multiple of 256, and 17 ("zeta") as a primitive 256th root
+ * of unity, the polynomial (X^256+1) splits in Z_q[X] into 128 irreducible
```

Comment:
remove 'in'

Reply:
No, "in" is correct as written, though perhaps "over" might be more standard math terminology here, a polynomial splits into factors (or not, perhaps irreducible) over the field of interest, depending on whether one is considering the smallest base field that contains the coëfficients, or some extension of that field.